command/agent/testagent.go

```diff
@@ -147,9 +148,13 @@ func (a *TestAgent) Start() *TestAgent {
 			panic(fmt.Sprintf("Error creating agent: %s", err))
 		}
 
-		agent.id = id
-		agent.LogOutput = a.LogOutput
+		logOutput := a.LogOutput
+		if logOutput == nil {
+			logOutput = os.Stderr
+		}
+		agent.LogOutput = logOutput
 		agent.LogWriter = a.LogWriter
+		agent.logger = log.New(logOutput, id, log.LstdFlags)
```

Comment:
I'll use the test name as an id as you suggested earlier. That makes more sense even if the logs are a bit misaligned. `grep TestName test.log` will then show you everything you need to know.